lib/beaker/dsl/install_utils/windows_utils.rb

```diff
@@ -141,6 +141,16 @@ def install_msi_on(hosts, msi_path, msi_opts = {}, opts = {})
             # therefore, this command will always exit 0 if either service is installed
             on host, Command.new("sc query puppet || sc query pe-puppet", [], { :cmdexe => true })
 
+            # (PA-514) value for PUPPET_AGENT_STARTUP_MODE should be present in
+            # registry and honored after install/upgrade.
+            reg_query_command = %Q(reg query "HKLM\\SOFTWARE\\Wow6432Node\\Puppet Labs\\PuppetInstaller" /v "RememberedPuppetAgentStartupMode" | findstr #{msi_opts['PUPPET_AGENT_STARTUP_MODE']})
+            on host, Command.new(reg_query_command, [], { :cmdexe => true })
+
+            start_mode = msi_opts['PUPPET_AGENT_STARTUP_MODE'] == "Automatic" ? "Auto" : msi_opts['PUPPET_AGENT_STARTUP_MODE']
```

Comment:
@MosesMendoza I was confused about the valid key; I thought that the valid key in `msi_opts` should be `Auto` instead of `Automatic`, and thus the yardoc for the method would need a change. However, I see now that it is defined locally as `start_mode` and thus there is no need to for this change.